src/odemis/driver/smaract.py

```diff
@@ -1522,6 +1528,11 @@ def _doReference(self, future):
 
                 logging.info("Referencing successful.")
 
+            if self._pos_deactive_after_ref and self._deactive_pos_after_ref is not None and self._is_referenced():
```

Comment:
Having two variables with almost the same names is really confusing. I'd recommend to not have a dedicated `_deactive_pos_after_ref` variable. Instead, directly read it in the metadata:
`if self._pos_deactive_after_ref and model.MD_FAV_POS_DEACTIVE in self._metadata ...`